Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -1019,7 +1028,23 @@ checkMissingDocs dgs esgs edgs efgs = do
         -- 2. Realise Globs.
         let realGlob t =
               concatMap globMatches
-                <$> mapM (runDirFileGlobM ops "") t
+                <$> mapM
+                  ( \pathOrGlob ->
+                      case pathOrGlob of
+                        Left filepath -> do
+                          case mpackageOps of
+                            Nothing ->
+                              pure []
+                            Just packageOps -> do
+                              exist <- doesFileExist packageOps filepath
+                              pure $
+                                if exist
+                                  then [GlobMatch filepath]
+                                  else [] -- TODO: probably should signal failure here
```

Comment:
On the one hand, we'll have already reported a problem if the filepath literal glob didn't match something previously. This check is trying to see if there are any other notable files present in these globs. So this triggering an error condition is unlikely: we'd need to have, like, `extra-source-files: README.md` and then `README.md` not being present in the directory.

I think this is redundant, so it's probably fine to have `[]` here.

A future improvement will remove the `realGlob` function from this file and accept the `[GlobResult FilePath]` directly as arguments, removing the redundant call to `runDirFileGlobM`.